lib/vault/api/auth.rb

```diff
@@ -186,6 +186,60 @@ def aws_ec2(role, pkcs7, nonce = nil)
       return secret
     end
 
+    # Authenticate via IAM EC2 method by providing a AWS CredentialProvider (either ECS, AssumeRole, etc.)
```

Comment:
It would be better to say something like "Authenticate via AWS IAM auth method" as this doesn't use the ec2 auth method at all.